lib/msal-node/src/client/ManagedIdentitySources/AzureArc.ts

```diff
@@ -64,13 +82,38 @@ export class AzureArc extends BaseManagedIdentitySource {
     }
 
     public static getEnvironmentVariables(): Array<string | undefined> {
-        const identityEndpoint: string | undefined =
+        let identityEndpoint: string | undefined =
             process.env[
                 ManagedIdentityEnvironmentVariableNames.IDENTITY_ENDPOINT
             ];
-        const imdsEndpoint: string | undefined =
+        let imdsEndpoint: string | undefined =
             process.env[ManagedIdentityEnvironmentVariableNames.IMDS_ENDPOINT];
 
+        // if either of the identity or imds endpoints are undefined, check if the himds executable exists
+        if (!identityEndpoint || !imdsEndpoint) {
+            // get the expected Windows or Linux file path of the himds executable
+            const fileDetectionPath: string =
+                AZURE_ARC_FILE_DETECTION[process.platform as keyof FilePathMap];
+            try {
```

Comment:
add logging statements for these logic branches

Reply:
The logging exists already on line 142